components/calendly_v2/calendly_v2.app.mjs

```diff
@@ -1,11 +1,213 @@
+  propDefinitions: {
+    user: {
+      type: "string",
+      label: "User UUID",
+      description: "An user UUID",
+    },
+    eventId: {
+      type: "string",
+      label: "Event ID",
+      description: "An event UUID",
+      async options({ prevContext }) {
+        return await this._makeAsyncOptionsRequest({
+          prevContext,
+          requestType: "listEvents",
+        });
+      },
+    },
+    eventType: {
+      type: "string",
+      label: "Event Type",
+      description: "An event type UUID",
+      async options({ prevContext }) {
+        return await this._makeAsyncOptionsRequest({
+          prevContext,
+          requestType: "listEventTypes",
+        });
+      },
+    },
+    inviteeEmail: {
+      type: "string",
+      label: "Inviteee Email",
+      description: "The invitee's email",
+      optional: true,
+    },
+    status: {
+      type: "string",
+      label: "Event Status",
+      description: "Whether the scheduled event is `active` or `canceled`",
+      optional: true,
+      options: [
+        "active",
+        "canceled",
+      ],
+    },
+    maxEventCount: {
+      type: "integer",
+      label: "Max Event Count",
+      description: "The max number of events that can be scheduled using this scheduling link",
+    },
+    paginate: {
+      type: "boolean",
+      label: "Paginate",
+      description: "Whether to paginate or not",
+      optional: true,
+    },
+    maxResults: {
+      type: "integer",
+      label: "Max Results",
+      description: "The number of rows to return",
+      optional: true,
+    },
+  },
   methods: {
-    // this.$auth contains connected account data
-    authKeys() {
-      console.log(Object.keys(this.$auth));
+    _baseUri() {
+      return "https://api.calendly.com";
+    },
+    _buildUserUri(user) {
+      return `${this._baseUri()}/users/${user}`;
+    },
+    _buildEventType(eventType) {
+      return `${this._baseUri()}/event_types/${eventType}`;
+    },
+    _getDefaultResponse() {
+      return {
+        collection: [],
+        pagination: {
+          count: 0,
+        },
+      };
+    },
+    _extractNextPageToken(nextPage) {
+      return nextPage
+        ? nextPage.split("page_token=")[1].split("&")[0]
+        : null;
+    },
+    _makeRequestOpts(opts) {
+      const path = opts.path ?? "";
+      const method = opts.method ?? "get";
+      const params = opts.params ?? {};
+      const headers = {
+        "Content-Type": "application/json",
+        "Authorization": `Bearer ${this.$auth.oauth_access_token}`,
+      };
+      return {
+        url: this._baseUri() + path,
+        method,
+        headers,
+        params,
+      };
+    },
+    async _makeAsyncOptionsRequest({
+      prevContext,
+      requestType,
+    }) {
+      const defaultParams = {
+        count: 20,
+      };
+      const { nextPageParameters = defaultParams } = prevContext;
+      let response = await this[requestType](nextPageParameters);
```

Comment:
This can be a constant since it's not being redefined.
```suggestion
      const response = await this[requestType](nextPageParameters);
```

Reply:
Issue addressed